demos/embeddings/watch-your-step-cora-demo.ipynb

```diff
@@ -11,9 +11,11 @@
     "from stellargraph.layer.watch_your_step import WatchYourStep, get_embeddings\n",
```

Comment:
<p>Not related to this PR specifically, but why does this need <code>steps_per_epoch</code> ? Isn't that basically implicitly controlled by the length of <code>train_gen</code> ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/1031/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Reply:
its a repeated tensorflow dataset so it needs `steps_per_epoch` specified, otherwise you get:

````ValueError: When passing an infinitely repeating dataset, you must specify the `steps_per_epoch` argument.```